### ISSUE TITLE
error in  trying to get mAP and Localization Error for detector_evaluation

### ISSUE THREAD

**Issue by lishen565** [rpautrat/SuperPoint#32]
hi ,when I try to follow the "detector_evaluation_magic_point.ipynb", I meet the trouble:
![default](https://user-images.githubusercontent.com/1812302/50899246-7edd1980-144c-11e9-9657-133126bc7b76.png)
I don't know where the "keypoint_map" comes from.Does it mean the detection of the superpoint over standed image? Please help me,thanks a lot !
![2019-01-09 20-29-36](https://user-images.githubusercontent.com/1812302/50899767-1000c000-144e-11e9-8846-2d7fff6b749c.png)

**Comment by rpautrat** [comment]
Hi!
`keypoint_map` is the ground truth for the feature points (it is generated at the same time as you generate the synthetic images).
The original information of the image (the image itself and the ground truth) are copied to the dictionary containing the prediction only when the name of the image is not present in the original data. It may be that it is not the case here, so can you just replace the lines 70-71 of `export_detections.py`:
```
if not ('name' in d):
    p.update(d) # Can't get the data back from the filename --> dump
```
by 
```p.update(d)```

And then run again `export_detections.py` for each experiment.

**Comment by lishen565** [comment]
Thanks a lot. I will follow your ideas and try again.

**Comment by lishen565** [comment]
I want to ask you another question,how can I get the keypoint_map of HPatches? Thanks a lot.

**Comment by rpautrat** [comment]
You can't, they don't exist. That's why training a neural net on real images for features detection is so hard: there is no ground truth.

**Comment by cumtchenchang** [comment]
@lishen565  Thank you for your question,  I am very interested in this problem. Have you got  mAP and Localization Error for detector_evaluation? Thank you.

**Comment by rpautrat** [comment]
@cumtchenchang, mAP and Localization Error does not make sense since there is no ground truth for feature point detectors on COCO or HPatches. It was only possible to compute them on the synthetic dataset where we created our own ground truth.

**Comment by cumtchenchang** [comment]
@rpautrat  Thank you for your response.  How to evaluate mAP and Localization Error on the synthetic dataset.  I followed your work and generated synthetic_shapes_v6 data, but I did not find other tutorials about how to compare on this data. In addition, is it more accurate to compare homography estimation on this data than HPatches? It is possibly not fair to compare homography between SuperPoint, SIFT, and ORB on different features. (In your work, extract SuperPoint detector and descriptor, ORB detector and descriptor, SIFT detector and descriptor).
